apollo-router/src/query_planner/bridge_query_planner_pool.rs

```diff
@@ -0,0 +1,200 @@
+use std::collections::HashMap;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use std::time::Instant;
+
+use apollo_compiler::validation::Valid;
+use async_channel::bounded;
+use async_channel::Sender;
+use futures::future::BoxFuture;
+use opentelemetry::KeyValue;
+use router_bridge::planner::Planner;
+use tokio::sync::oneshot;
+use tokio::task::JoinSet;
+use tower::Service;
+use tower::ServiceExt;
+
+use super::bridge_query_planner::BridgeQueryPlanner;
+use super::QueryPlanResult;
+use crate::error::QueryPlannerError;
+use crate::error::ServiceBuildError;
+use crate::services::QueryPlannerRequest;
+use crate::services::QueryPlannerResponse;
+use crate::spec::Schema;
+use crate::Configuration;
+
+static CHANNEL_SIZE: usize = 10_000;
```

Comment:
Should we consider making this configurable? A user could still end up in a bad place with a router which has no effective pushback.
Also: If not configurable, which is fine, this should be a lot lower than 10_000. Let's imagine it takes 20ms to plan a query. (not untypical). Then a queue of 1_000 means a maximum wait of 20seconds... Even if it was 1ms, that's still a potential 1 second wait. 10_000 could mean a 3 min+ wait...

Perhaps consider a queue size of 1_000 at most.

Reply:
Great point, 10_000 was chosen arbitrarily at first. 1000 would make sense. Configurable could potentially be useful, at the same time if you hit the max channel size here it's unlikely configuration is the thing you'd want to address. A sane upper limit sounds good for now?

Reply:
1_000 sounds good to me. also because we know of the current queue's length, we could followup to make the service not ready if the length reaches a certain treshold!

Reply:
Fixed in [1f7e478](https://github.com/apollographql/router/pull/4897/commits/1f7e478781f24f12f064ca669cddd8688b82db6d)